src/radio.rs

```diff
@@ -161,23 +159,62 @@ impl Phy {
             state: State::Idle,
         };
 
-        crate::crit_sect::locked(|cs| {
-            let prev_isr_data = ISR_DATA.borrow(cs).replace(Some(isr_data));
-            assert!(prev_isr_data.is_none());
-        });
+        let isr_data = Mutex::new(RefCell::new(isr_data));
 
-        Self {
-            isr_data: &ISR_DATA,
-        }
+        Self { isr_data }
+    }
+
+    fn get_global_instance_raw() -> &'static OnceCell<Phy> {
+        static PHY: OnceCell<Phy> = OnceCell::new();
```

Comment:
It should be enough to keep `IsrData` as `static`. The rest of `Phy` can be kept by the module creating given instance.
```suggestion
        static ISR_DATA: OnceCell<Mutex<RefCell<IsrData>>> = OnceCell::new(); 
```

and in `new_global_instance()`:
```
self.isr_data = get_global_instance_raw();
```